### PR TITLE
Add arrow key navigation in the main menu

### PR DESCRIPTION
**Pull request by razovi**
Hi, this is my first pull request. I'm a complete amateur so I would love to hear your opinions on how I can improve my code. If this feature is something you want I would be open to reimplementing it with your suggestions. As it is now there are some weird interactions between mouse hovering and keyboard navigation. Also, I couldn't get the code formatter tool to work, sorry.

### CODE REVIEW

**Comment by toadsworth-jr** [comment]
What exactly does this code plan to do? Does pressing the left and right arrow keys allow you to select different options in the main menu?

**Comment by razovi** [comment]
Yes, exactly. However, it only cycles between the buttons to the right of the osu logo.

File: osu.Game/Screens/Menu/MainMenuButton.cs [hunk 1]
```diff
@@ -156,6 +157,15 @@ protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint,
...
 
+        public bool OnNavigation()
+        {
+            return this.OnHover(new HoverEvent(new InputState()));
+        }
+
+        public void OnNavigationLost()
+        {
+            this.OnHoverLost(new HoverLostEvent(new InputState()));
+        }
```
Comment (frenzibyte) [inline osu.Game/Screens/Menu/MainMenuButton.cs:168]: This is not the right way of marking a button as hovered. Have a look at `VolumeOverlay` and `SelectionCycleFillFlowContainer`, which provide something similar to what you're trying to achieve.

**Review by frenzibyte — CHANGES_REQUESTED** [review]
Thanks for your contribution. The feature seems OK but the code behind it requires a refactor to reach an acceptable and legible state. I've left a note on a potential direction that you can follow through.

**Comment by peppy** [comment]
Even if we were to accept this change, there would need to be tests.

**Pull request closed without merging**